src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -145,6 +145,8 @@ protected function initialize()
         }
         foreach ($this->resources as $key => $params) {
             list($format, $resource, $locale, $domain) = $params;
+            if($locale==null)
```

Comment:
It is better to check with an identical comparison, and even better use native PHP functions like:
```PHP
if (\is_null($locale)) {}
```

Reply:
In Symfony code we always use `null === $locale`. But in any case I agre with @fabpot that we should rather investigate why `$locale` is `null` here in the first place and fix that.

Reply:
Hmm, but why identical comparison better then native functions? I think readability is the same...

Reply:
@xabbuh I'm just interesting know this from you :D

Reply:
The comparison is faster.

Reply:
> Hmm, but why identical comparison better then native functions? I think readability is the same...

Maybe you should read this article https://en.wikipedia.org/wiki/Yoda_conditions#Advantage and this https://symfony.com/doc/current/contributing/code/standards.html#structure

Reply:
@rand0mdev I know about Yoda-style..
@derrabus Thanks! :)

Reply:
In PHP 7 and higher, `\is_null()` is a special function and it's replaced by an opcode, so it should be as fast as the comparison (see https://github.com/php/php-src/blob/80d3ce3d19cb3df5ad9802d852a6bee831cb03ab/Zend/zend_compile.c#L4193)

I think the main reason is that in Symfony code we try to avoid calls to `if (empty($someVar))` and we always prefer to be explicit: (e.g. `if (null !== $someVar && '' !== $someVar)`). So, always using `null !== $someVar` is more consistent with the rest of the code.

Reply:
@javiereguiluz That is correct, if you fully-qualify the `is_null()` call as you did. In addition to what you said, Symfony maintains a consistent code-style over all branches, so as long as we have 3.4 with php 5.5 support, the comparison is the saver bet. :-)